cli/cmd/trigger_delivery_test.go

```diff
@@ -206,3 +206,65 @@ func TestCheckImageNonAvailabilityD(t *testing.T) {
 		}
 	}
 }
+
+// TestTriggerDeliveryNonExistingService tests the trigger delivery
+// with non-existing service.
+func TestTriggerDeliveryNonExistingService(t *testing.T) {
+
+	credentialmanager.MockAuthCreds = true
+	checkEndPointStatusMock = true
+
+	receivedEvent := make(chan bool)
+	mocking = true
+	ts := httptest.NewServer(
+		http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			w.Header().Add("Content-Type", "application/json")
+			w.WriteHeader(200)
+			if strings.Contains(r.RequestURI, "v1/project") {
+				project := &apimodels.Project{
+					ProjectName: "sockshop",
+					Stages: []*apimodels.Stage{
+						{
+							StageName: "dev",
+						},
+					},
+				}
+				marshal, _ := json.Marshal(project)
+				w.Write(marshal)
+			}
+			if strings.Contains(r.RequestURI, "v1/event") {
+				defer r.Body.Close()
+				bytes, err := ioutil.ReadAll(r.Body)
+				if err != nil {
+					t.Errorf("could not read received event payload: %s", err.Error())
+				}
+				event := &apimodels.KeptnContextExtendedCE{}
+				if err := json.Unmarshal(bytes, event); err != nil {
+					t.Errorf("could not decode received event: %s", err.Error())
+				}
+				if *event.Type != keptnv2.GetTriggeredEventType("dev.artifact-delivery") {
+					t.Errorf("did not receive correct event: %s", err.Error())
+				}
+				go func() {
+					receivedEvent <- true
+				}()
+			} else if strings.Contains(r.RequestURI, "/v1/metadata") {
+				defer r.Body.Close()
+				w.Write([]byte(metadataMockResponse))
+				return
+			}
+			return
+		}),
+	)
+	defer ts.Close()
+
+	os.Setenv("MOCK_SERVER", ts.URL)
```

Comment:
is this necessary?

Reply:
yes, we get errors without it.

Reply:
what errors do you get?

Reply:
I will probably re-write the test to be more general :) Hopefully it will be better afterwards :)

Reply:
Yeah and actually for a heads-up: I get segmentation fault regarding some missig variable/nil variable. Did not debug it until now but I can look at it afterwards.

Reply:
I have generalized the test, but I am receiving failures for this test on github, locally works fine. Seems to me like some kind of infrastructure problem, debugging it.